x/evm/keeper/funtoken_from_coin.go

```diff
@@ -75,16 +77,43 @@ func (k *Keeper) deployERC20ForBankCoin(
 	if err != nil {
 		return gethcommon.Address{}, errors.Wrap(err, "failed to pack ABI args")
 	}
-	bytecodeForCall := append(embeds.SmartContract_ERC20Minter.Bytecode, packedArgs...)
+	input := append(embeds.SmartContract_ERC20Minter.Bytecode, packedArgs...)
 
-	// nil address for contract creation
-	evmResp, _, err := k.CallContractWithInput(
-		ctx, evm.EVM_MODULE_ADDRESS, nil, true, bytecodeForCall, Erc20GasLimitDeploy,
+	evmMsg := gethcore.NewMessage(
+		evm.EVM_MODULE_ADDRESS,
+		nil, /*contract*/
+		k.GetAccNonce(ctx, evm.EVM_MODULE_ADDRESS),
+		big.NewInt(0), /*amount*/
+		Erc20GasLimitDeploy,
+		big.NewInt(0), /*gasFeeCap*/
+		big.NewInt(0), /*gasTipCap*/
+		big.NewInt(0), /*gasPrice*/
+		input,
+		gethcore.AccessList{},
+		false, /*isFake*/
+	)
+	evmCfg := k.GetEVMConfig(ctx)
+	txConfig := k.TxConfig(ctx, gethcommon.BigToHash(big.NewInt(0)))
+	stateDB := k.Bank.StateDB
+	if stateDB == nil {
+		stateDB = k.NewStateDB(ctx, txConfig)
+	}
+	evmObj := k.NewEVM(ctx, evmMsg, evmCfg, nil /*tracer*/, stateDB)
+	evmResp, err := k.CallContractWithInput(
+		ctx, evmObj, evm.EVM_MODULE_ADDRESS, nil, true /*commit*/, input, Erc20GasLimitDeploy,
 	)
 	if err != nil {
 		k.ResetGasMeterAndConsumeGas(ctx, ctx.GasMeter().Limit())
 		return gethcommon.Address{}, errors.Wrap(err, "failed to deploy ERC20 contract")
 	}
+
+	err = stateDB.Commit()
+	if err != nil {
+		return gethcommon.Address{}, errors.Wrap(err, "failed to commit stateDB")
+	}
+	// Don't need the StateDB anymore because it's not usable after committing
+	k.Bank.StateDB = nil
```

Comment:
We always have to set the `Bank.StateDB` to nil so that the StateDB doesn't accrue pending changes from the BeginBlocker/EndBlocker hooks (e.g. distribution and fee_collector balance changes).

Plus, there's no guarantee that StateDB is usable after `Commit()` is called, so it's better to just let the Golang garbage collector dispose of it.